src/components/TopicViews.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import PreviewCard from './PreviewCard';
+import { RootState } from '../store/reducers';
+import { ResourcesSlice } from '../store/resourcesSlice';
+import TopicHeader from './TopicHeader';
+
+interface TopicViewProps {
+  topicId: number, 
+}
+
+function TopicViews( { topicId }: TopicViewProps) {
+  const allTopics = useSelector((state: RootState) => state.topics); 
+  const currentTopic = allTopics[topicId]; 
+  const name = currentTopic.name;
+
+  const selectTopicResources = (state: RootState) =>
+    Object.keys(state.resources as ResourcesSlice)
+      .filter(
+        (id) =>
+          state.resources[(id as unknown) as number].tags.includes(name, 0) &&
+          !state.resources[(id as unknown) as number].tags.includes('Troubleshooting', 0))
+      .reduce<ResourcesSlice>((res, key) => {
+        res[(key as unknown) as number] =
+          state.resources[(key as unknown) as number];
+        return res;
+      }, {});
+
+  const selectTroubleshootingResources = (state: RootState) =>
+    Object.keys(state.resources as ResourcesSlice)
+      .filter(
+        (id) =>
+          state.resources[(id as unknown) as number].tags.includes(name, 0) &&
+          state.resources[(id as unknown) as number].tags.includes('Troubleshooting', 0))
+      .reduce<ResourcesSlice>((res, key) => {
+        res[(key as unknown) as number] =
+          state.resources[(key as unknown) as number];
+        return res;
+      }, {});
+
+  const topics = useSelector(selectTopicResources);
```

Comment:
The naming here is a little confusing — `topics` sounds like it's a list of topics, but it's actually just all the non-troubleshooting resources for a single topic. Maybe we could call these two something like `guideResources` and `troubleshootingResources`?